coaddAnalysis.py

```diff
@@ -0,0 +1,234 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+from astropy.visualization import (ZScaleInterval, SqrtStretch, ImageNormalize)
+
+import lsst.geom as geom
+from lsst.pipe.base import Struct
+
+__all__ = ("findPatches", "plotTractOutline", "mosaicCoadd")
+
+
+def findPatches(butler, skymap, verbose=True):
+    # Instead of using a manually defined patch list to figure out which patches
+    # have warps/coadds in them, go through each patch and try opening its coadd.
+    # If that succeeds, add the patch ID to our happy patch list.
+
+    for tractInfo in skymap:
+        patch_list = [str(patch.getIndex()[0]) + ',' + str(patch.getIndex()[1]) for patch in tractInfo]
+        good_patch_list = []
+        for patch in patch_list:
+            try:
+                dataId = {'filter': 'g', 'tract': 0, 'patch': patch}
+                butler.get('deepCoadd_visitInfo', dataId=dataId)
+            except:
+                if verbose:
+                    print('No data in this patch', patch)
+                continue
+            else:
+                if verbose:
+                    print('This patch has a coadd', patch)
+                good_patch_list.append(patch)
+    return good_patch_list
+
+
+def percent(values, p=0.5):
+    """Return a value a faction of the way between the min and max values in a list."""
+    m = min(values)
+    interval = max(values) - m
+    return m + p*interval
+
+
+def bboxToRaDec(bbox, wcs):
+    """Get the corners of a BBox and convert them to lists of RA and Dec."""
```

Comment:
I think this is already built into the WCS and doesn't need its own function, e.g.
```
skyCorners = [wcs.pixelToSky(pixPos) for pixPos in bbox.getCorners()]
```
Note that you can subsequently do
```
patchList = tractInfo.findPatchList(skyCorners)
```